modal/_container_io_manager.py

```diff
@@ -621,13 +622,34 @@ async def memory_restore(self) -> None:
                     "CUDA device availability may be inaccurate."
                 )
 
+        # Unpatch socket.close() to restore default TIME_WAIT behavior.
+        # Refer modal._contaienr_utils.patch_socket_close() for full exetent of patch.
+        if is_socket_patched():
+            try:
+                unpatch_socket_close()
+                logger.debug("socket.close() unpatched successfully")
+            except Exception as exc:
+                logger.warning(f"failed to unpatch socket.close(): {exc}")
+
         self._client = await _Client.from_env()
 
     async def memory_snapshot(self) -> None:
         """Message server indicating that function is ready to be checkpointed."""
         if self.checkpoint_id:
             logger.debug(f"Checkpoint ID: {self.checkpoint_id} (Memory Snapshot ID)")
 
+        if connections := get_open_connections():
```

Comment:
Similar to what was worked through with `async with self._heartbeat_condition:` can this be structured as a context manager and not two separate blocks. It'll be more readable and maintainable that way. 

https://github.com/modal-labs/modal-client/pull/2004/commits/c0f5fcbcd934514ca15c9973f81cb4c2794187a1